src/plot_api/helpers.js

```diff
@@ -120,17 +119,17 @@ exports.cleanLayout = function(layout) {
             }
             delete ann.ref;
         }
+
         cleanAxRef(ann, 'xref');
         cleanAxRef(ann, 'yref');
     }
 
-    if(layout.shapes !== undefined && !Array.isArray(layout.shapes)) {
-        Lib.warn('Shapes must be an array.');
-        delete layout.shapes;
-    }
     var shapesLen = (layout.shapes || []).length;
```

Comment:
`Array.isArray` again... I guess if you test this situation you may find even more of these...